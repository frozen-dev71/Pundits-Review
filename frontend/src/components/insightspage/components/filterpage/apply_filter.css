/*
// ======================================
// CSS for Apply Filter Button
// ======================================
*/

.apply-filter-button-wrapper {
	width: 100vw;
	position: fixed;
	bottom: 10vh;
	display: flex;
	justify-content: center;
}

.apply-filter-button {
	background-color: white;
    bottom: 10vh;
    border: 1px solid black;
    box-shadow: 4px 4px 10px grey;
    letter-spacing: .2em;
    padding: .75em;
    cursor: pointer;

}