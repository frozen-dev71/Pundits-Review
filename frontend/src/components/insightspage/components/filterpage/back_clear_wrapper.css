/*
// ======================================
// CSS for the Back Clear Wrapper
// ======================================
*/


.filter-page-back-clear-wrapper {

	display: flex;
	width: 90vw;
	justify-content: space-between;
	margin: 1.5em 0;
}


/*
// ======================================
// MEDIA QUERIES
// ======================================
*/

@media (min-width: 800px) {

	.filter-page-back-clear-wrapper {
		width: 80vw;
	}

}


@media (min-width: 1200px) {

	.filter-page-back-clear-wrapper {
		width: 70vw;
	}

}



@media (min-width: 1600px) {

	.filter-page-back-clear-wrapper {
		width: 60vw;
	}

}