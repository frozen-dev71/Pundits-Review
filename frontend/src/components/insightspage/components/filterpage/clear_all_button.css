/*
// ======================================
// CSS for the Filter Clear All Button
// ======================================
*/


.clear-all-button {
	
	display: flex;
    align-items: center;
    justify-content: center;
    font-size: .75em;
    letter-spacing: .1em;
    background-color: white;
    border: none;
    background-color: black;
    padding: .5em;
    color: white;
    cursor: pointer;
}