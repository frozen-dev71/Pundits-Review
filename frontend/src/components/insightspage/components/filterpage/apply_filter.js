// ======================================
// Apply Filter Component - button to trigger new filter action
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";

// Local JS Imports
import { applyFilterPlayerReviewScores, applyFilterPlayerTotalScores, applyFilterPlayerAvgScores, setFilterActive, setGridCardView } from '../../../actions/actions';

// CSS Imports 
import './apply_filter.css';

// ===========
// Apply Filter Component
// ===========


export class ApplyFilterButtonWrapper extends Component {

	_handleClick = () => {

		let filterSettings = {

			position: this.props.positionFilters,
			club: this.props.clubFilters,
			nationality: this.props.nationalityFilters

		}
		
		// Apply the filter to the current player review scores array
		this.props.applyFilterPlayerReviewScores(this.props.originalPlayerReviewScores, filterSettings);

		// Apply the filter to the player total reviews array
		this.props.applyFilterPlayerTotalScores(this.props.originalPlayerTotalReviews, filterSettings);

		// Apply the filter to the player avg scores array
		this.props.applyFilterPlayerAvgScores(this.props.originalPlayerAvgScores, filterSettings);

		// Deactivate the filter page
		this.props.setFilterActive(false)

		// Set Grid View - Insights page
		this.props.setGridCardView("CARD");

	}

	render() {

		let result = (

				<div className="apply-filter-button-wrapper">
					<button className="apply-filter-button body-text-light" onClick={this._handleClick}>
						APPLY FILTER
					</button>
				</div>

			);


		return result;
	}
}


// ===========
// Connect component to redux store
// ===========
const mapStateToProps = state => {
  return { 

  		originalPlayerReviewScores: state.originalPlayerReviewScores,
  		playerReviewScores: state.playerReviewScores,
  		originalPlayerTotalReviews: state.originalPlayerTotalReviews,
  		playerTotalReviews: state.playerTotalReviews,
  		originalPlayerAvgScores: state.originalPlayerAvgScores,
  		playerAvgScores: state.playerAvgScores,
  		filterActive: state.filterActive,
  		positionFilters: state.positionFilters,
		clubFilters: state.clubFilters,
		nationalityFilters: state.nationalityFilters

  	};
};


export const ApplyFilterButton = connect(mapStateToProps, { applyFilterPlayerReviewScores, applyFilterPlayerTotalScores, applyFilterPlayerAvgScores, setGridCardView, setFilterActive })(ApplyFilterButtonWrapper)
