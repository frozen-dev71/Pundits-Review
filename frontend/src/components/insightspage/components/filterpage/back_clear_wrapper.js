// ======================================
// Back & Clear Wrapper Element - contains back and clear buttons on filter page
// ======================================


// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";

// Local JS Imports
import { FilterBackButton } from './filter_back_button';
import { ClearAllButton } from './clear_all_button';


// CSS Imports
import './back_clear_wrapper.css';



// ===========
// Back Clear Wrapper Component 
// ===========


export class BackClearWrapper extends Component {

	render() {

		let result = (

				<div className="filter-page-back-clear-wrapper">

					<FilterBackButton />
					<ClearAllButton />

				</div>

			);


		return result;
	}
}