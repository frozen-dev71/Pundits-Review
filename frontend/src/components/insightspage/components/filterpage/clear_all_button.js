// ======================================
// Clear All Button Component - Allows user to clear all previously specified filters
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";

// Local JS Imports 
import { clearAllFilters } from '../../../actions/actions';

// CSS Imports
import './clear_all_button.css';


// ===========
// Clear All Button Component 
// ===========

export class ClearAllButtonWrapper extends Component {

	_handleClick = () => {

		// Clear all filters
		this.props.clearAllFilters();

		// Remove display of all ticks 
		document.querySelectorAll('.filter-tick-container').forEach(function(element) {
			element.className = "filter-tick-container";
		})

	}

	render() {

		let result = (

				<button className="clear-all-button body-text-light" onClick={this._handleClick}>
					CLEAR ALL
				</button>

			);

		return result;

	}
}



// ===========
// Connect component to redux store
// ===========
const mapStateToProps = state => {
  return { 

  		nationalityFilters: state.nationalityFilters,
  		clubFilters: state.clubFilters,
  		positionFilters: state.positionFilters

  	};
};


export const ClearAllButton = connect(mapStateToProps, {clearAllFilters})(ClearAllButtonWrapper);

