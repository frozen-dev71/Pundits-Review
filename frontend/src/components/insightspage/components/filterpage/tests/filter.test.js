// ======================================
// This file tests for the components listed in the INSIGHTSPAGE / FILTER sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that ApplyFilterButton renders correctly
// 2. Check that BackClearWrapper renders correctly
// 3. Check that ClearAllButton renders correctly
// 4. Check that ClubFilter renders correctly
// 5. Check that ClubFilterElement renders correctly
// 6. Check that FilterBackButton renders correctly
// 7. Check that FilterPage renders correctly
// 8. Check that NationalityFilter renders correctly
// 9. Check that NationalityFilterElement renders correctly
// 10. Check that PlusMinus renders correctly
// 11. Check that PositionFilter renders correctly
// 12. Check that PositionFilterElement renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

// Local JS Imports
import { ApplyFilterButton } from '.././apply_filter';
import { BackClearWrapper } from '.././back_clear_wrapper';
import { ClearAllButton } from '.././clear_all_button';
import { ClubFilter } from '.././club_filter';
import { ClubFilterElement } from '.././club_filter_element';
import { FilterBackButton } from '.././filter_back_button';
import { FilterPage } from '.././filterpage';
import { NationalityFilter } from '.././nationality_filter';
import { NationalityFilterElement } from '.././nationality_filter_element';
import { PlusMinus } from '.././plusminus';
import { PositionFilter } from '.././position_filter';
import { PositionFilterElement } from '.././position_filter_element';


import store from '../../../../store/store';

// Configurations
Enzyme.configure({ adapter: new Adapter() })

// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" & 'Warning: React.createElement: type is invalid -- expected a string (for built-in components)' - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('Filter Page Component Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server') || msg.includes('Warning: React.createElement: type is invalid -- expected a string (for built-in components)')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });


// End of referenced code

  	// ===========
	// TEST 1. Check that ApplyFilterButton renders correctly
	// ===========

	describe('ApplyFilterButton', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ApplyFilterButton/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 2. Check that BackClearWrapper renders correctly
	// ===========

	describe('BackClearWrapper', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><BackClearWrapper/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 3. Check that ClearAllButton renders correctly
	// ===========

	describe('ClearAllButton', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ClearAllButton/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that ClubFilter renders correctly
	// ===========

	describe('ClubFilter', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ClubFilter/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 5. Check that ClubFilterElement renders correctly
	// ===========

	describe('ClubFilterElement', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><ClubFilterElement/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 6. Check that FilterBackButton renders correctly
	// ===========

	describe('FilterBackButton', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><FilterBackButton/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 7. Check that FilterPage renders correctly
	// ===========

	describe('FilterPage', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><FilterPage/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 8. Check that NationalityFilter renders correctly
	// ===========

	describe('NationalityFilter', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><NationalityFilter/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 9. Check that NationalityFilterElement renders correctly
	// ===========

	describe('NationalityFilterElement', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><NationalityFilterElement/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 10. Check that PlusMinus renders correctly
	// ===========

	describe('PlusMinus', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><PlusMinus/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 11. Check that PositionFilter renders correctly
	// ===========

	describe('PositionFilter', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><PositionFilter/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 12. Check that PositionFilterElement renders correctly
	// ===========

	describe('PositionFilterElement', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><PositionFilterElement/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


});
