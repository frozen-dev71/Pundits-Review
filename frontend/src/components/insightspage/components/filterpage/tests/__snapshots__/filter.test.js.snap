// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Page Component Tests ApplyFilterButton should render correctly 1`] = `
<div
  class="apply-filter-button-wrapper"
>
  <button
    class="apply-filter-button body-text-light"
  >
    APPLY FILTER
  </button>
</div>
`;

exports[`Filter Page Component Tests BackClearWrapper should render correctly 1`] = `
<div
  class="filter-page-back-clear-wrapper"
>
  <button
    class="filter-back-button-container"
  >
    <i
      aria-hidden="true"
      class="fa fa-angle-left filter-back-button-icon"
    />
    <p
      class="filter-back-button-text body-text-light"
    >
      BACK
    </p>
  </button>
  <button
    class="clear-all-button body-text-light"
  >
    CLEAR ALL
  </button>
</div>
`;

exports[`Filter Page Component Tests ClearAllButton should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(ClearAllButtonWrapper) />
  </Provider>
</Router>
`;

exports[`Filter Page Component Tests ClubFilter should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <ClubFilter />
  </Provider>
</Router>
`;

exports[`Filter Page Component Tests ClubFilterElement should render correctly 1`] = `
<button
  class="club-filter-element"
>
  <div
    class="filter-tick-container"
  >
    <i
      aria-hidden="true"
      class="fa fa-check filter-tick"
    />
  </div>
  <span
    class="undefined club-badge club-filter-element-club-badge"
  />
  <p
    class="club-filter-element-text body-text-light undefined"
  />
</button>
`;

exports[`Filter Page Component Tests FilterBackButton should render correctly 1`] = `
<button
  class="filter-back-button-container"
>
  <i
    aria-hidden="true"
    class="fa fa-angle-left filter-back-button-icon"
  />
  <p
    class="filter-back-button-text body-text-light"
  >
    BACK
  </p>
</button>
`;

exports[`Filter Page Component Tests FilterPage should render correctly 1`] = `
<div
  class="filter-page-wrapper"
>
  <div
    class="filter-page-back-clear-wrapper"
  >
    <button
      class="filter-back-button-container"
    >
      <i
        aria-hidden="true"
        class="fa fa-angle-left filter-back-button-icon"
      />
      <p
        class="filter-back-button-text body-text-light"
      >
        BACK
      </p>
    </button>
    <button
      class="clear-all-button body-text-light"
    >
      CLEAR ALL
    </button>
  </div>
  <div
    class="filter-item-wrapper"
  >
    <button
      aria-required="true"
      class="filter-bar-dropdown position-filter-bar background-light"
    >
      <div
        class="plus-minus-container"
        id="position-container-plus-minus"
      >
        <div
          class="plus-minus-bar plus-minus-bar--vertical"
        />
        <div
          class="plus-minus-bar plus-minus-bar--horizontal plus"
          id="position-bar-plus-minus"
        />
      </div>
      <p
        class="filter-bar-title-text body-text-light"
      >
        POSITION
      </p>
    </button>
    <div
      class="position-filter-elements-wrapper"
    >
      <div
        class="position-filter-elements-container"
      >
        <button
          class="position-filter-element"
        >
          <p
            class="position-filter-element-text body-text-light"
          >
            GK
          </p>
          <div
            class="filter-tick-container"
            id="position-gk-tick"
          >
            <i
              aria-hidden="true"
              class="fa fa-check filter-tick"
            />
          </div>
        </button>
        <button
          class="position-filter-element"
        >
          <p
            class="position-filter-element-text body-text-light"
          >
            DF
          </p>
          <div
            class="filter-tick-container"
            id="position-df-tick"
          >
            <i
              aria-hidden="true"
              class="fa fa-check filter-tick"
            />
          </div>
        </button>
      </div>
      <div
        class="position-filter-elements-container"
      >
        <button
          class="position-filter-element"
        >
          <p
            class="position-filter-element-text body-text-light"
          >
            MD
          </p>
          <div
            class="filter-tick-container"
            id="position-md-tick"
          >
            <i
              aria-hidden="true"
              class="fa fa-check filter-tick"
            />
          </div>
        </button>
        <button
          class="position-filter-element"
        >
          <p
            class="position-filter-element-text body-text-light"
          >
            FWD
          </p>
          <div
            class="filter-tick-container"
            id="position-fwd-tick"
          >
            <i
              aria-hidden="true"
              class="fa fa-check filter-tick"
            />
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    class="filter-item-wrapper"
  >
    <button
      aria-required="true"
      class="filter-bar-dropdown position-filter-bar background-light"
    >
      <div
        class="plus-minus-container"
        id="club-container-plus-minus"
      >
        <div
          class="plus-minus-bar plus-minus-bar--vertical"
        />
        <div
          class="plus-minus-bar plus-minus-bar--horizontal plus"
          id="club-bar-plus-minus"
        />
      </div>
      <p
        class="filter-bar-title-text body-text-light"
      >
        CLUB
      </p>
    </button>
    <div
      class="club-filter-element-wrapper"
    >
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="bournemouth-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="bournemouth-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light bournemouth-color"
        >
          Bournemouth
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="arsenal-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="arsenal-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light arsenal-color"
        >
          Arsenal
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="villa-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="villa-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light villa-color"
        >
          Aston Villa
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="brighton-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="brighton-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light brighton-color"
        >
          Brighton
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="burnley-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="burnley-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light burnley-color"
        >
          Burnley
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="chelsea-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="chelsea-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light chelsea-color"
        >
          Chelsea
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="palace-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="palace-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light palace-color"
        >
          C. Palace
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="everton-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="everton-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light everton-color"
        >
          Everton
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="leicester-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="leicester-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light leicester-color"
        >
          Leicester
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="liverpool-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="liverpool-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light liverpool-color"
        >
          Liverpool
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="city-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="city-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light city-color"
        >
          Man City
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="united-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="united-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light united-color"
        >
          Man Utd
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="newcastle-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="newcastle-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light newcastle-color"
        >
          Newcastle
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="norwich-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="norwich-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light norwich-color"
        >
          Norwich
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="sheffield-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="sheffield-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light sheffield-color"
        >
          Sheffield Utd
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="southampton-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="southampton-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light southampton-color"
        >
          Southampton
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="spurs-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="spurs-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light spurs-color"
        >
          Spurs
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="watford-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="watford-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light watford-color"
        >
          Watford
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="westham-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="westham-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light westham-color"
        >
          West Ham
        </p>
      </button>
      <button
        class="club-filter-element"
      >
        <div
          class="filter-tick-container"
          id="wolves-badge-club-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
        <span
          class="wolves-badge club-badge club-filter-element-club-badge"
        />
        <p
          class="club-filter-element-text body-text-light wolves-color"
        >
          Wolves
        </p>
      </button>
    </div>
  </div>
  <div
    class="filter-item-wrapper"
  >
    <button
      aria-required="true"
      class="filter-bar-dropdown position-filter-bar background-light"
    >
      <div
        class="plus-minus-container"
        id="nationality-container-plus-minus"
      >
        <div
          class="plus-minus-bar plus-minus-bar--vertical"
        />
        <div
          class="plus-minus-bar plus-minus-bar--horizontal plus"
          id="nationality-bar-plus-minus"
        />
      </div>
      <p
        class="filter-bar-title-text body-text-light"
      >
        NATIONALITY
      </p>
    </button>
    <div
      class="nationality-filter-element-wrapper"
    />
  </div>
  <div
    class="apply-filter-button-wrapper"
  >
    <button
      class="apply-filter-button body-text-light"
    >
      APPLY FILTER
    </button>
  </div>
</div>
`;

exports[`Filter Page Component Tests NationalityFilter should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(NationalityFilterWrapper) />
  </Provider>
</Router>
`;

exports[`Filter Page Component Tests NationalityFilterElement should render correctly 1`] = `
<button
  class="nationality-filter-element"
>
  <div
    class="filter-tick-container"
  >
    <i
      aria-hidden="true"
      class="fa fa-check filter-tick"
    />
  </div>
  <span
    class="undefined nationality-filter-element-flag"
  />
  <p
    class="nationality-filter-element-text body-text-light"
  />
</button>
`;

exports[`Filter Page Component Tests PlusMinus should render correctly 1`] = `
<div
  class="plus-minus-container"
>
  <div
    class="plus-minus-bar plus-minus-bar--vertical"
  />
  <div
    class="plus-minus-bar plus-minus-bar--horizontal plus"
  />
</div>
`;

exports[`Filter Page Component Tests PositionFilter should render correctly 1`] = `
<div
  class="filter-item-wrapper"
>
  <button
    aria-required="true"
    class="filter-bar-dropdown position-filter-bar background-light"
  >
    <div
      class="plus-minus-container"
      id="position-container-plus-minus"
    >
      <div
        class="plus-minus-bar plus-minus-bar--vertical"
      />
      <div
        class="plus-minus-bar plus-minus-bar--horizontal plus"
        id="position-bar-plus-minus"
      />
    </div>
    <p
      class="filter-bar-title-text body-text-light"
    >
      POSITION
    </p>
  </button>
  <div
    class="position-filter-elements-wrapper"
  >
    <div
      class="position-filter-elements-container"
    >
      <button
        class="position-filter-element"
      >
        <p
          class="position-filter-element-text body-text-light"
        >
          GK
        </p>
        <div
          class="filter-tick-container"
          id="position-gk-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
      </button>
      <button
        class="position-filter-element"
      >
        <p
          class="position-filter-element-text body-text-light"
        >
          DF
        </p>
        <div
          class="filter-tick-container"
          id="position-df-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
      </button>
    </div>
    <div
      class="position-filter-elements-container"
    >
      <button
        class="position-filter-element"
      >
        <p
          class="position-filter-element-text body-text-light"
        >
          MD
        </p>
        <div
          class="filter-tick-container"
          id="position-md-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
      </button>
      <button
        class="position-filter-element"
      >
        <p
          class="position-filter-element-text body-text-light"
        >
          FWD
        </p>
        <div
          class="filter-tick-container"
          id="position-fwd-tick"
        >
          <i
            aria-hidden="true"
            class="fa fa-check filter-tick"
          />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Filter Page Component Tests PositionFilterElement should render correctly 1`] = `
<button
  class="position-filter-element"
>
  <p
    class="position-filter-element-text body-text-light"
  />
  <div
    class="filter-tick-container"
  >
    <i
      aria-hidden="true"
      class="fa fa-check filter-tick"
    />
  </div>
</button>
`;
